builtin/mainmenu/tab_online.lua

```diff
@@ -285,14 +316,46 @@ local function search_server_list(input)
 	-- Search the serverlist
 	local search_result = {}
 	for i, server in ipairs(serverlistmgr.servers) do
-		local name_matches, description_matches = true, true
-		for _, keyword in ipairs(keywords) do
+		local name_matches, description_matches, filter_matches = true, true, true
+
+		-- Check if keyword found
+		for _, keyword in ipairs(query.keywords) do
 			name_matches = name_matches and not not
 					(server.name or ""):lower():find(keyword, 1, true)
 			description_matches = description_matches and not not
 					(server.description or ""):lower():find(keyword, 1, true)
 		end
-		if name_matches or description_matches then
+
+		-- Check if mods found
+		local mods_lower = {}
```

Comment:
This can go if you already do the lowercasing in the query parsing. Same below.

Reply:
Why?
`mods_lower` are the mods from the server list, not from the search query.

Reply:
I saw the inconsistent / missing `:lower()` in the parsing and wrongly concluded that this would be the query, sorry.

Indeed it's a good idea to preprocess it this way, though for mods it might not be necessary, but don't quote me on that, best to do it anyways.

If we're already doing this, may I suggest something like the following (optional):

```lua
local function lowercase_set(t)
    local set = {}
    for _, v in ipairs(t) do
        set[v:lower()] = true
    end
    return set
end
```

then you can just do something like

```lua
local server_mods = lowercase_set(server.mods or {})
for _, mod in ipairs(query.mods) do
    if not server_mods[mod] then
        return false
    end
end
```

which could again be extracted into a function (basically it's `all(server_mods[mod] for mod in query.mods)` in Python terms, but we don't have `all` here either).

---

(Side note: If we had some basic FP / iterator abstractions like `map`, `all` this could be much nicer, something like `local server_mods = table.set(table.map(server.mods, string.lower))`.)

Reply:
I did put all the lowercasing into a `uncapitalize_server` function, to preprocess it. This should be easier to understand, I think.
(It gets executed only once for every server.)

Also, there was still a bug, I though, I already changed this, the gameid must be lowered too. e.g. search for `game:exile`